Adds routing, a controller, and a view for an HOC Students Served dashboard.

ashercodeorg opened the pull request:
In doing so, separates HOC related routes from admin_reports_controller to admin_hoc_controller.

ashercodeorg commented:
Hold off on reviewing, test failure is legitimate.

ashercodeorg commented:
Good to go, review at your leisure.

Hamms commented:
It looks like the `hoc_signups` method wasn't actually removed from AdminReports; is that correct? If so, why?

Hamms commented on dashboard/app/controllers/admin_hoc_controller.rb:
```diff
@@ -0,0 +1,66 @@
+
+  def students_served
+    SeamlessDatabasePool.use_persistent_read_connection do
+      @data = Properties.get(:hoc_metrics)
```
where is this being generated, again?

Hamms commented on dashboard/app/views/admin_hoc/students_served.html.haml:
```diff
@@ -0,0 +1,47 @@
+          %td
+            %span= number_with_delimiter(count)
+- else
+  Error, @data.present? is false.
```
Could we get a more user-parsable error message?

Hamms commented on dashboard/config/routes.rb:
```diff
@@ -182,7 +182,10 @@ def redirect_to_teacher_dashboard
 
   # one-off internal reports
   get '/admin/temp/diversity_survey', to: 'admin_reports#diversity_survey', as: 'diversity_survey'
-  get '/admin/temp/hoc_signups', to: 'admin_reports#hoc_signups', as: 'hoc_signups'
```
Always love seeing lines with 'temp' in red

Hamms commented on dashboard/app/controllers/admin_hoc_controller.rb:
```diff
@@ -0,0 +1,66 @@
+      data_2014 = DB[:forms].
+                  where('kind = ? AND created_at > ? AND created_at < ?', 'HocSignup2014', '2014-08-01', '2015-01-01').
+                  group(:name, :email).
+                  # TODO(asher): Is this clumsy notation really necessary? Is Sequel
```
Now that you've had a bit more time to play with Sequel since you originally wrote this, do you have a satisfying answer to this question?

Hamms replied:
Or rather, have you done any more Sequel since you originally wrote this?

ashercodeorg commented:
Deleted hoc_signups from admin_controller (oops, nice catch).

Properties.get(:hoc_metrics) is set by the bin/cron/analyze_hoc_activity (see: https://github.com/code-dot-org/code-dot-org/blob/staging/bin/cron/analyze_hoc_activity#L97).

Added a slightly more friendly error message (contact an engineer).

Let me look at the SQL, if I can simplify, I'll do so in another PR.

Hamms commented:
Awesome! LGTM

ashercodeorg commented:
Test failure seems spurious:

RackAttackTest
  test_limits                                                     PASS (0.01s)
  test_property_limits_enforced                                   FAIL (0.08s)
Minitest::Assertion:         4th property set is rate limited..
        Expected: 429
          Actual: 200
        /home/ubuntu/code-dot-org/shared/test/test_rack_attack.rb:163:in `test_property_limits_enforced'
        /home/ubuntu/code-dot-org/shared/test/test_helper.rb:60:in`block (2 levels) in around'
        /home/ubuntu/.rvm/gems/ruby-2.2.3/gems/sequel-4.10.0/lib/sequel/database/transactions.rb:119:in `_transaction'
        /home/ubuntu/.rvm/gems/ruby-2.2.3/gems/sequel-4.10.0/lib/sequel/database/transactions.rb:100:in`block in transaction'
        /home/ubuntu/.rvm/gems/ruby-2.2.3/gems/sequel-4.10.0/lib/sequel/database/connecting.rb:229:in `block in synchronize'
        /home/ubuntu/.rvm/gems/ruby-2.2.3/gems/sequel-4.10.0/lib/sequel/connection_pool/threaded.rb:104:in`hold'
        /home/ubuntu/.rvm/gems/ruby-2.2.3/gems/sequel-4.10.0/lib/sequel/database/connecting.rb:229:in `synchronize'
        /home/ubuntu/.rvm/gems/ruby-2.2.3/gems/sequel-4.10.0/lib/sequel/database/transactions.rb:89:in`transaction'
        /home/ubuntu/code-dot-org/shared/test/test_helper.rb:59:in `block in around'
        /home/ubuntu/.rvm/gems/ruby-2.2.3/gems/vcr-3.0.0/lib/vcr/util/variable_args_block_caller.rb:9:in`call'
        /home/ubuntu/.rvm/gems/ruby-2.2.3/gems/vcr-3.0.0/lib/vcr/util/variable_args_block_caller.rb:9:in `call_block'
        /home/ubuntu/.rvm/gems/ruby-2.2.3/gems/vcr-3.0.0/lib/vcr.rb:188:in`use_cassette'
        /home/ubuntu/code-dot-org/shared/test/test_helper.rb:58:in `around'
